xmake/modules/detect/sdks/find_vstudio.lua

```diff
@@ -210,11 +207,14 @@ function main()
         -- * version > 15.0 eschews registry entries; but `vswhere` (included with version >= 15.2) can be used to find VC install path
         -- ref: https://github.com/Microsoft/vswhere/blob/master/README.md @@ https://archive.is/mEmdu
         local vswhere_VCAuxiliaryBuildDir = nil
-        if ((version + 0) >= 15) and vswhere then
-            local vswhere_vrange = format("%s,%s)", version, (version+1))
-            local result = os.iorunv(vswhere.program, {"-property", "installationpath", "-products", "Microsoft.VisualStudio.Product.BuildTools", "-version", vswhere_vrange})
-            if result then 
-                vswhere_VCAuxiliaryBuildDir = result:trim() .."\\VC\\Auxiliary\\Build" 
+        if ((version + 0) >= 15) then
+            local vswhere = find_tool("vswhere")
+            if vswhere then
+                local vswhere_vrange = format("%s,%s)", version, (version+1))
+                local result = os.iorunv(vswhere.program, {"-property", "installationpath", "-products", "Microsoft.VisualStudio.Product.BuildTools", "-version", vswhere_vrange})
+                if result then
```

Comment:
But this condition `if ((version + 0) >= 15) then` will always be satisfied. 

`find_tool("vswhere")` will always be executed once or even multiple times if the version list contains 16.0 17.0 .. in future.

Reply:
The condition should only be satisfied if the MSVC/VS version being tested for is greater than or equal to "15.0". If that's not correct, I need to rework it so that it is... the entire "vswhere" logic only applies to MSVC/VS versions at/after version 15.0/2017.

Reply:
Ah, I see your point. It's only being used for the "15.0" version now, but in the future, the search will be run multiple times because it's not being cached. And it'll always run at least once because the search always includes "15.0".

Logical failure on my part.

I remove that commit and re-push it.

Reply:
```lua
-- always only execute once
local vswhere = find_tool("vswhere") 
for _, version in ipairs({"17.0", "16.0", "15.0", "14.0", "12.0", "11.0", "10.0", "9.0", "8.0", "7.1", "7.0", "6.0", "5.0", "4.2"}) do
    -- ...
end
```

```lua
for _, version in ipairs({"17.0", "16.0", "15.0", "14.0", "12.0", "11.0", "10.0", "9.0", "8.0", "7.1", "7.0", "6.0", "5.0", "4.2"}) do
    -- will be executed three times
    if ((version + 0) >= 15) then
        local vswhere = find_tool("vswhere") 
    end
end
```

```lua
for _, version in ipairs({"15.0", "14.0", "12.0", "11.0", "10.0", "9.0", "8.0", "7.1", "7.0", "6.0", "5.0", "4.2"}) do
    -- always only execute once
    if ((version + 0) >= 15) then
        local vswhere = find_tool("vswhere") 
    end
end
```

Reply:
ok